flexget/plugins/clients/flood.py

```diff
@@ -0,0 +1,446 @@
+import base64
+import os
+import time
+import urllib.parse
+
+from loguru import logger
+from requests import Response
+from requests.exceptions import RequestException
+
+from flexget import plugin
+from flexget.entry import Entry
+from flexget.event import event
+from flexget.plugin import DependencyError, PluginError
+from flexget.task import Task
+
+try:
+    from jwt import decode
+
+    imported = True
+except ImportError:
+    imported = False
+
+logger = logger.bind(name='flood')
+
+
+class FloodClient:
+    @staticmethod
+    def request(
+        task: Task, config: dict, method: str, url: str, success_list: list, **kwargs
+    ) -> Response:
+        """
+        Send a request to Flood.
+        Returns None if the response status_code is not in the success_list.
+        """
+
+        try:
+            response = task.requests.request(
+                method, urllib.parse.urljoin(config["url"], url), **kwargs
+            )
+        except RequestException as e:
+            raise PluginError(
+                f"An error occurred while attempting to send a request to Flood: {e}"
+            )
+
+        if not response.status_code in success_list:
+            raise PluginError(
+                f"Failed to send request to Flood. Received status {response.status_code} expected one of {success_list}."
+            )
+
+        return response
+
+    @staticmethod
+    def is_jwt_expired(task: Task) -> bool:
```

Comment:
`is_jwt_valid` seems like a better name here, since it checks both existence and expiration.

Reply:
I like it, does make a lot more sense!

Reply:
Renamed `is_jwt_expired` to `is_jwt_invalid` in Winter/Flexget@b3cb221